dttools/src/stringtools.c

```diff
@@ -722,6 +722,38 @@ void string_replace_backslash_codes(const char *a, char *b)
 	*b = 0;
 }
 
+char *string_replace_percents( char *str, const char *replace )
+{
+	/* Common case: do nothing if no percents. */
+	if(!strchr(str,'%')) return str;
```

Comment:
This needs to be `return xxstrdup(str);`. Otherwise the caller may double-free `str` or try to free a stack allocated `str`.